src/test/java/org/example/Operations/QueryOperationsTest.java

```diff
@@ -82,7 +84,7 @@ public void testSupportForSpecificQueries() {
     }
 
     @Test
-    public void testAddingAndRemovingSpecificQueries() {
+    public void testAddingAndRemovingSpecificQueries() throws UnsupportedEncodingException {
```

Comment:
For the test, you can just assertNoThrow on the call. No need for the exception to bubble up to this level.

Reply:
Done.